examples/lpt_nbody_demo.py

```diff
@@ -42,19 +58,18 @@ def fttk(nc: int, sharding) -> list:
     -------
     list
         List of wave number arrays for each dimension.
-    """
+  """
   kd = np.fft.fftfreq(nc) * 2 * np.pi
-  return [
-      jax.make_array_from_callback(
-          (nc, 1, 1),
-          sharding=NamedSharding(sharding.mesh, P('z')),
-          data_callback=lambda x: kd.reshape([-1, 1, 1])[x]),
-      jax.make_array_from_callback(
-          (1, nc, 1),
-          sharding=NamedSharding(sharding.mesh, P(None, 'y')),
-          data_callback=lambda x: kd.reshape([1, -1, 1])[x]),
-      kd.reshape([1, 1, -1])
-  ]
+
+  @partial(
+      shmap,
```

Comment:
OK I have not thought of using shardmap as a way to distribute arrays
Come to think of it we can even send just the mesh size as input with a replicate in_spec ( P(None) ) and get back the right sharding

Reply:
nah you need to have the input spec like this such that it will slice correctly the input vector for each dimension.

Reply:
But I mean that we can use a sharded fftk function that takes a mesh_shape and output a sharded array
This solves some many problems for me in jaxPM

Reply:
sure